diff --git a/extension.js b/extension.js
index 3d6ed9d..e94ff4e 100644
--- a/extension.js
+++ b/extension.js
@@ -404,17 +404,24 @@ class MprisLabel extends PanelMenu.Button {
   
 			const blacklist = ALBUM_BLACKLIST.toLowerCase().replaceAll(' ','').split(',');
 			if(!blacklist.includes(this.player.identity.toLowerCase()))
-				this.icon = this.player.getArtUrlIcon(size, ICON_PLACE, ICON_PADDING);
+				this.icon = this.player.getArtUrlIcon(size);
 		}
 
-		if(this.icon == null)
-			this.icon = this.player.getIcon(ICON_PLACE, SYMBOLIC_ICON, ICON_PADDING);
+		if(this.icon == null){
+			this.icon = this.player.getIcon(SYMBOLIC_ICON);
+			if (SYMBOLIC_ICON)
+				this.icon.set_style('-st-icon-style: symbolic;');
+		}
 
 		if (this.icon != null | undefined){
-			if (ICON_PLACE == "right")
+			if (ICON_PLACE == "right"){
+				this.icon.set_style(this.icon.get_style() + "padding-left: " + ICON_PADDING + "px;padding-right: 0px;");
 				this.box.add_child(this.icon);
-			else if (ICON_PLACE == "left")
+			}
+			else if (ICON_PLACE == "left"){
+				this.icon.set_style(this.icon.get_style() + "padding-left: 0px;padding-right: " + ICON_PADDING + "px;");
 				this.box.insert_child_at_index(this.icon,0);
+			}
 		}
 	}
 
diff --git a/players.js b/players.js
index ba01ec8..20f1f4f 100644
--- a/players.js
+++ b/players.js
@@ -235,50 +235,26 @@ class Player {
 		}
 		return ""
 	}
-	getArtUrlIcon(size, iconPlace, padding){
-		const shellVersion = Number.parseInt(imports.misc.config.PACKAGE_VERSION.split('.'));
+	getArtUrlIcon(size){
 		const url = this.stringFromMetadata("mpris:artUrl",this.metadata);
 
-		let icon_left_padding = 0;
-		let icon_right_padding = 0;
-		if (shellVersion >= 3)
-			if (iconPlace == "right")
-				icon_left_padding = padding
-			else if (iconPlace == "left")
-				icon_right_padding = padding
-
 		if(url.length>0)
 			this.albumArt = new St.Icon({
 				gicon: Gio.Icon.new_for_string(url),
 				style_class: 'system-status-icon',
 				icon_size: size,
-				style: "padding: 0px; padding-left: " + icon_left_padding + "px;padding-right: " + icon_right_padding + "px;"
+				style: "padding: 0px;"
 			})
 		else this.albumArt = null;
 
 		return this.albumArt
 	}
-	getIcon(iconPlace, symbolicIcon, padding){
-		const shellVersion = Number.parseInt(imports.misc.config.PACKAGE_VERSION.split('.'));
-
-		let icon_left_padding = 0;
-		let icon_right_padding = 0;
-		if (shellVersion >= 3)
-			if (iconPlace == "right")
-				icon_left_padding = padding
-			else if (iconPlace == "left")
-				icon_right_padding = padding
-
+	getIcon(){
 		let icon = new St.Icon({
 			style_class: 'system-status-icon',
-			fallback_icon_name: 'audio-volume-high',
-			style: "padding-left: " + icon_left_padding + "px;padding-right: " + icon_right_padding + "px;"
+			fallback_icon_name: 'audio-volume-high'
 		});
 
-		if (symbolicIcon) {
-			icon.set_style(icon.get_style() + '-st-icon-style: symbolic;');
-		}
-
 		if(this.desktopApp == null | undefined)
 			return icon
 
